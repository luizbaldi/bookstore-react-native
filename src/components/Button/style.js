/* @flow */
import styled from 'styled-components/native'
import { Colors } from '../../resources'

export const Container = styled.TouchableOpacity`
  padding-horizontal: ${({ iconOnly }) => iconOnly ? 6 : 35};
  borderRadius: 50;
  flexDirection: row;
  alignItems: center;
  justifyContent: center;
  background-color: ${({ backgroundColor }) => backgroundColor};
  height: ${({ small }) => small ? 28 : 35};
`

export const Label = styled.Text`
  color: ${Colors.white};
  font-size: ${({ small }) => small ? 10 : 12};
`

export const Content = styled.View`
  flex-direction: row
`

export const Icon = styled.Image`

`