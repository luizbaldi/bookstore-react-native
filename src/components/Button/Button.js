import React from 'react';
import { ActivityIndicator } from 'react-native';

import { Colors } from '@/resources';

import { Container, Label, Content, Icon } from './style';

type Props = {
  label: string,
  onPress(): void,
  loading: boolean,
  small: boolean,
  icon: string,
  backgroundColor: string
};

const Button = ({
  label,
  onPress,
  loading,
  small,
  icon,
  backgroundColor
}: Props) => {
  const iconOnly = icon && !label;

  return (
    <Container
      onPress={onPress}
      small={small}
      iconOnly={iconOnly}
      backgroundColor={backgroundColor}
    >
      {loading ? (
        <ActivityIndicator size='small' color={Colors.white} />
      ) : (
        <Content>
          {Boolean(icon) && <Icon source={icon} />}
          <Label>{label}</Label>
        </Content>
      )}
    </Container>
  );
};

Button.defaultProps = {
  label: '',
  onPress: () => {},
  loading: false,
  small: false,
  icon: null,
  backgroundColor: Colors.blue
};

export default Button;
