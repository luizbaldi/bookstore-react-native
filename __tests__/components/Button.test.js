import React from 'react'
import Button from '../../src/components/Button'
import renderer from 'react-test-renderer'

test('should render button', () => {
  const buttonTree = renderer.create(<Button label='Potatoe' />).toJSON()
  expect(buttonTree).toMatchSnapshot()
})
