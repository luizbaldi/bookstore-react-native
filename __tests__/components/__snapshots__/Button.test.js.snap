// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3090e9",
      "borderRadius": 50,
      "flexDirection": "row",
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 35,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 12,
          },
        ]
      }
    >
      Potatoe
    </Text>
  </View>
</View>
`;
